src/router/Router.js

```diff
@@ -465,11 +480,31 @@ export default class Router extends DisposableBase {
         }
     }
     _halt(err) {
+        //Storing the current status locally
+        // as moveToHalted() may move the state onwards
+        // and we only want to run the error handlers on the first error
+        let localStatus = this._state.currentStatus;
```

Comment:
can we change this so the `Status.Halted` check happens here, then we give this variable name something more descriptive. 

For example:
```
let isInitialHaltingError = this._state.!== Status.Halted

// then later 

if(isInitialHaltingError) {
}
```
you probably don't need the comments then as it's fairly self descriptive (hopefully :)).